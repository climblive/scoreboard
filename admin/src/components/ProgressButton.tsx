import { CircularProgress } from "@material-ui/core";
import Button, { ButtonProps } from "@material-ui/core/Button";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import React, { memo } from "react";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    wrapper: {
      position: "relative",
      display: "inline-block",
    },
    buttonProgress: {
      position: "absolute",
      top: "50%",
      left: "50%",
      marginTop: -12,
      marginLeft: -12,
    },
    button: {
      width: "100%",
    },
  })
);

interface Props {
  loading?: boolean;
}

export const ProgressButton = memo((props: Props & ButtonProps) => {
  const classes = useStyles();
  const { loading, ...rest } = props;

  return (
    <div className={classes.wrapper}>
      <Button
        {...rest}
        disabled={props.disabled || props.loading}
        className={classes.button}
      >
        {props.children}
      </Button>
      {props.loading && (
        <CircularProgress size={24} className={classes.buttonProgress} />
      )}
    </div>
  );
});
