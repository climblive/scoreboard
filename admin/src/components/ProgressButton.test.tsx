import React from "react";
import renderer from "react-test-renderer";
import { ProgressButton } from "./ProgressButton";

test("Render normal button", () => {
  const component = renderer.create(
    <ProgressButton variant="contained" color="primary">
      Hello World
    </ProgressButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Render fullwidth button", () => {
  const component = renderer.create(
    <ProgressButton fullWidth>Hello World</ProgressButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Render loading button", () => {
  const component = renderer.create(
    <ProgressButton loading>Hello World</ProgressButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
