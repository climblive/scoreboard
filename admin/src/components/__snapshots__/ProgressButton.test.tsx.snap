// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render fullwidth button 1`] = `
<div
  className="makeStyles-wrapper-1 makeStyles-wrapper-5"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-fullWidth"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Hello World
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Render loading button 1`] = `
<div
  className="makeStyles-wrapper-1 makeStyles-wrapper-6"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Hello World
    </span>
  </button>
  <div
    className="MuiCircularProgress-root makeStyles-buttonProgress-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 24,
        "width": 24,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`Render normal button 1`] = `
<div
  className="makeStyles-wrapper-1 makeStyles-wrapper-4"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Hello World
    </span>
  </button>
</div>
`;
